### DISCUSSION TITLE
Periods at the end of single line comments

### DISCUSSION THREAD

**Discussion by sebastienros** [OrchardCMS/OrchardCore#14199] [General]
I think it's ridiculous to enforce periods at the end of single line comments. I believe the convention should be to add a dot only if there is a second sentence.

Specifically in code comments, not method documentation.

`/// <summary>Foo.</summary>` is fine
`// TODO: do something.` is not

**Comment by Piedone** [comment]
Oh, I love myself some [bike shedding](https://en.wikipedia.org/wiki/Law_of_triviality)!

Documentation, in any form, is English, and you (should) write full sentences, so IMO the English grammar applies: use punctuation.

**Reply by jtkech** [reply]
Yes these discussions are good to define step by step and when we want:

1. Rules that we want to remove.
2. Recommendations without adding rules => Nothing to do or only cleaning files gradually.
3. Rules that we want to add (maybe only suggestions for now, not errors or warnings).

Yes this rule has been added with 4 other ones when `StyleCopAnalyzers` were introduced.

- Personally I now prefer to have a period at the end of a single line comment, but maybe just a new habit so I'm fine to not have this rule, can be easily removed, let me know.

- For example we now have the rule (as a suggestion) that a comment should begin with a space, I like it.

- I also prefer that a comment starts with a capital letter but I didn't see this rule, only for documentation. So maybe only part of `2.` if we agree with the recommendation.

About `2.` when it's only a recommendation maybe we should not update all the files in one pass but gradually.

When I worked on rules I also applied some implicit recommendations but mainly in files that were updated to fix current rule violations (the main goal).  Also I didn't add any new rule, only removed one, see https://github.com/OrchardCMS/OrchardCore/issues/14198#issuecomment-1696475396.

**Reply by jtkech** [reply]
Just saw that the current rule is

    dotnet_diagnostic.SA1629.severity = suggestion  # Documentation text should end with a period.

So it is only applied to class/method/field Documentation texts.

Currently it is not applied to simple comments (single line or not).

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1629.md

**Comment by sebastienros** [comment]
Then I'd suggest not to use PRs to add periods to it, it's adding noise and making it more difficult for reviewers to focus on the actual changes.

**Reply by jtkech** [reply]
Okay I agree.

Oups maybe I already did it in the PR I'm working on but not too much.
